Sources/Sentry/SentryANRTrackingIntegration.m

```diff
@@ -34,7 +34,19 @@ @implementation SentryANRTrackingIntegration
 
 - (void)installWithOptions:(SentryOptions *)options
 {
-    if ([self shouldBeDisabled:options]) {
+    if (![self shouldBeEnabled:@[
+            [[SentryOptionWithDescription alloc] initWithOption:options.enableAppHangTracking
+                                                     optionName:@"enableAppHangTracking"],
+            [[SentryOptionWithDescription alloc]
+                initWithOption:(options.appHangTimeoutInterval > 0)
+                           log:@"Not going to enable App Hanging integration because "
+                               @"appHangTimeoutInterval is 0."],
+            [[SentryOptionWithDescription alloc]
+                initWithOption:(![SentryDependencyContainer.sharedInstance
+                                       .crashWrapper isBeingTraced])
+                           log:@"Not going to enable App Hanging integration because the debugger "
+                               @"is attached."],
+        ]]) {
```

Comment:
Im not sure about this. Looks like we're not improving too much, almost the same number of lines than before. And I think the `removeEnabledIntegration` part could also be automatic. 

What do you think about using an enum collection for integration options, each integration has the enum list that make sense for it, and every message and check is centralised in the 'BaseIntegration'.

Reply:
> almost the same number of lines than before

Not inside of the individual integration files though. Overall, yes, but that is because the new base class is added, new tests, etc.

> I think the removeEnabledIntegration part could also be automatic.

True

> What do you think about using an enum collection for integration options, each integration has the enum list that make sense for it, and every message and check is centralised in the 'BaseIntegration'.

That could be a good idea indeed.